app/frontend/util/ChampaignAPI.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import _ from 'lodash';
+import $ from 'jquery';
+import { camelizeKeys } from './util';
+
+export type OperationResponse = {
+  success: boolean,
+  errors: ?{[id:string]: string[]}
+}
+
+const parseResponse = (response):OperationResponse => {
```

Comment:
`parseResponse` feels unnecessary since any 200-299 response would resolve and any > 400 response would reject.